### PR TITLE
 #23331 Add parent project field to database

### PR DESCRIPTION
**Pull request by mc2contributor**
Provide support for sub-projects, as requested in issue #23331, "Nested Projects".
This adds the required database field.

### CODE REVIEW

**Comment by mc2contributor** [comment]
Should the new field for the parent project ID be 'parent', 'fk_projet', or something else?

**Comment by eldy** [comment]
> Should the new field for the parent project ID be 'parent', 'fk_projet', or something else?

Field should be   fk_project

Note that this represent a very important work due to credentials on projets and tasks. That may need a long time because such a feature is stable so to not break existing features, i recommend to implement all code (once this PR on structure is ok with name of field) with a condition 
` if (getDolGlobalInt('PROJECT_SUPPORT_SUB_PROJECT'))`

File: htdocs/install/mysql/migration/17.0.0-18.0.0.sql [hunk 1]
```diff
@@ -411,3 +411,4 @@ CREATE TABLE llx_c_invoice_subtype (
 
 ALTER TABLE llx_c_invoice_subtype ADD UNIQUE INDEX uk_c_invoice_subtype (entity, code);
 
+ALTER TABLE llx_fk_project ADD COLUMN parent int DEFAULT NULL;
```
Comment (eldy) [inline htdocs/install/mysql/migration/17.0.0-18.0.0.sql:414]: There is no table llx_fk_project. fk_project must be field name not table name

Comment (mc2contributor) [reply htdocs/install/mysql/migration/17.0.0-18.0.0.sql:414]: Whoops! Thanks for catching that.

**Comment by mc2contributor** [comment]
Added "Status" column to child projects list to more closely fit the requirements of the feature request.

**Comment by eldy** [comment]
According to Contributing rules. Changes on database structure must be done in a separated and standalone PR.
Code that use the new structure into another one.

**Comment by mc2contributor** [comment]
PR revised to include only the database structure change.

**Pull request merged**